google-cloud-spanner/acceptance/spanner/client/params/float32_test.rb

```diff
@@ -0,0 +1,116 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "spanner_helper"
+
+describe "Spanner Client", :params, :float32, :spanner do
+  let(:db) { spanner_client }
+
+  it "queries and returns a float32 parameter" do
+    results = db.execute_query "SELECT @value AS value", params: { value: 12.0 }, types: { value: :FLOAT32 }
```

Comment:
As discussed, please add a new test for `SELECT CAST(@value AS FLOAT32) AS value` with the type code not set. This should return a `FLOAT32: 12.0` (approximately)

Reply:
Thanks for spotting this case. Added another test (just below this test).